File: apps/api/src/main/java/com/lifttrack/api/domain/models/WorkoutSessionExerciseSet.java
package com.lifttrack.api.domain.models;

import java.math.BigDecimal;
import java.time.Instant;
import java.util.UUID;

public record WorkoutSessionExerciseSet(
        UUID uuid,
        UUID workoutSessionExerciseUuid,
        int setNumber,
        int reps,
        BigDecimal weight,
        Integer rir,
        boolean warmup,
        Instant createdAt
) {
}

File: apps/web/src/domain/models/workout-session-exercise-set.ts
export interface WorkoutSessionExerciseSet {
  readonly uuid: string
  readonly workoutSessionExerciseUuid: string
  readonly setNumber: number
  readonly reps: number
  readonly weight: number
  readonly rir: number | null
  readonly warmup: boolean
  readonly createdAt: string
}

export interface CreateWorkoutSessionExerciseSet {
  readonly setNumber: number
  readonly reps: number
  readonly weight: number
  readonly rir?: number
  readonly warmup: boolean
}
